server/datastore/mysql/operating_system_vulnerabilities_test.go

```diff
@@ -354,57 +355,166 @@ func testDeleteOutOfDateOSVulnerabilities(t *testing.T, ds *Datastore) {
 	kernel3 := fleet.Software{Name: "linux-image-8.11.0-11-generic", Version: "8.11.0-11.11", Source: "deb_packages", IsKernel: true}
 	software := []fleet.Software{
 		kernel1,
 		kernel2,
-		kernel3, // this one has 0 vulns
+		kernel3, // this one will have 0 vulns
 	}
-	_, err = ds.UpdateHostSoftware(ctx, host.ID, software)
-	require.NoError(t, err)
-	require.NoError(t, ds.LoadHostSoftware(ctx, host, false))
 
-	cpes := []fleet.SoftwareCPE{
-		{SoftwareID: host.Software[0].ID, CPE: "somecpe"},
-	}
-	_, err = ds.UpsertSoftwareCPEs(ctx, cpes)
-	require.NoError(t, err)
-	require.NoError(t, ds.LoadHostSoftware(ctx, host, false))
-
-	vulns := []fleet.SoftwareVulnerability{
-		{SoftwareID: host.Software[0].ID, CVE: "CVE-2022-0001"},
-		{SoftwareID: host.Software[1].ID, CVE: "CVE-2022-0002"},
-		// {SoftwareID: host.Software[2].ID, CVE: "CVE-2022-0002"},
-	}
-	for _, v := range vulns {
-		_, err = ds.InsertSoftwareVulnerability(ctx, v, fleet.NVDSource)
-		require.NoError(t, err)
+	cases := []struct {
+		name                 string
+		team                 bool
+		host                 *fleet.Host
+		software             []fleet.Software
+		vulns                []fleet.SoftwareVulnerability
+		vulnsByKernelVersion map[string][]string
+		os                   fleet.OperatingSystem
+	}{
+		{
+			name:  "ubuntu no team",
+			team:  false,
+			host:  test.NewHost(t, ds, "host_ubuntu2410", "", "hostkey_ubuntu2410", "hostuuid_ubuntu2410", time.Now(), test.WithPlatform("linux")),
+			vulns: []fleet.SoftwareVulnerability{{CVE: "CVE-2025-0001"}, {CVE: "CVE-2025-0002"}, {CVE: "CVE-2025-0003"}},
+			vulnsByKernelVersion: map[string][]string{
+				kernel1.Version: {"CVE-2025-0001", "CVE-2025-0002"},
+				kernel2.Version: {"CVE-2025-0003"},
+				kernel3.Version: nil,
+			},
+			software: software,
+			os:       fleet.OperatingSystem{Name: "Ubuntu", Version: "24.10", Arch: "x86_64", KernelVersion: "6.11.0-9-generic", Platform: "ubuntu"},
+		},
+		{
+			name:     "ubuntu with team",
+			team:     true,
+			host:     test.NewHost(t, ds, "host_ubuntu2404", "", "hostkey_ubuntu2404", "hostuuid_ubuntu2404", time.Now(), test.WithPlatform("linux")),
+			software: software[1:],
+			vulns:    []fleet.SoftwareVulnerability{{CVE: "CVE-2025-0004"}, {CVE: "CVE-2025-0005"}, {CVE: "CVE-2025-0003"}}, // Note the overlap; kernel2 has 0003 from the previous test
+			vulnsByKernelVersion: map[string][]string{
+				kernel2.Version: {"CVE-2025-0004", "CVE-2025-0005", "CVE-2025-0003"},
+				kernel3.Version: nil,
+			},
+			os: fleet.OperatingSystem{Name: "Ubuntu", Version: "24.04", Arch: "x86_64", KernelVersion: "6.11.0-9-generic", Platform: "ubuntu"},
+		},
+		{
+			name:     "amazon linux with team",
+			team:     true,
+			host:     test.NewHost(t, ds, "host_amzn2023", "", "hostkey_amzn2023", "hostuuid_amzn2023", time.Now(), test.WithPlatform("fedora")),
+			software: []fleet.Software{{Name: "Amazon Linux", Version: "2023", Arch: "x86_64", Source: "rpm_packages", IsKernel: true}},
```

Comment:
This should be named `kernel`, right? Also, for realism we want a real kernel version here, which 2023 isn't.

Reply:
Huh, I think I just misread something. It's definitely not `2023` 😅

Reply:
Fixed fedora as well, name should be `kernel-core`